src/pydvl/valuation/methods/semivalue.py

```diff
@@ -0,0 +1,142 @@
+
+A **semi-value** is any valuation function with the form:
+
+$$
+v_\text{semi}(i) = \sum_{i=1}^n w(k)
+                     \sum_{S \subset D_{-i}^{(k)}} [U(S_{+i})-U(S)],
+$$
+
+where $U$ is the utility, and the coefficients $w(k)$ satisfy the property:
+
+$$
+\sum_{k=1}^n w(k) = 1.
+$$
+
+This is the largest class of marginal-contribution-based valuation methods. These
+compute the value of a data point by evaluating the change in utility when the data
+point is removed from one or more subsets of the data.
+"""
+
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Any
+
+from joblib import Parallel, delayed
+
+from pydvl.utils.progress import Progress
+from pydvl.valuation.base import Valuation
+from pydvl.valuation.dataset import Dataset
+from pydvl.valuation.result import ValuationResult
+from pydvl.valuation.samplers import IndexSampler
+from pydvl.valuation.stopping import StoppingCriterion
+from pydvl.valuation.utility.base import UtilityBase
+from pydvl.valuation.utils import (
+    ensure_backend_has_generator_return,
+    make_parallel_flag,
+)
+
+__all__ = ["SemivalueValuation"]
+
+
+class SemivalueValuation(Valuation):
+    r"""Abstract class to define semi-values.
+
+    Implementations must only provide the `coefficient()` method, corresponding
+    to the semi-value coefficient.
+
+    !!! Note
+        For implementation consistency, we slightly depart from the common definition
+        of semi-values, which includes a factor $1/n$ in the sum over subsets.
+        Instead, we subsume this factor into the coefficient $w(k)$.
+        TODO: see ...
+
+    Args:
+        utility: Object to compute utilities.
+        sampler: Sampling scheme to use.
+        is_done: Stopping criterion to use.
+        progress: Whether to show a progress bar.
+    """
+
+    algorithm_name = "Semi-Value"
+
+    def __init__(
+        self,
+        utility: UtilityBase,
+        sampler: IndexSampler,
+        is_done: StoppingCriterion,
+        progress: dict[str, Any] | bool = False,
+    ):
+        super().__init__()
+        self.utility = utility
+        self.sampler = sampler
+        self.is_done = is_done
+        self.tqdm_args: dict[str, Any] = {
+            "desc": f"{self.__class__.__name__}: {str(is_done)}"
+        }
+        # HACK: parse additional args for the progress bar if any (we probably want
+        #  something better)
+        if isinstance(progress, bool):
+            self.tqdm_args.update({"disable": not progress})
+        else:
+            self.tqdm_args.update(progress if isinstance(progress, dict) else {})
+
+    @abstractmethod
+    def coefficient(self, n: int, k: int) -> float:
+        """Computes the coefficient for a given subset size.
+
+        Args:
+            n: Total number of elements in the set.
+            k: Size of the subset for which the coefficient is being computed
+        """
+        ...
+
+    def fit(self, data: Dataset):
```

Comment:
Why not change the signature to this to allow using the dataset set on the utility, if there is one of course?

```suggestion
    def fit(self, data: Dataset | None = None):
```